web/js/modules/map/util.js

```diff
@@ -146,110 +146,50 @@ export function getMapParameterSetup(
- * @return {array}                An extent array. Used to calculate
- * the extent for prev, next & current day
- */
-function calculateExtent(layerExtent, map) {
-  const viewportExtent = map.getView().calculateExtent();
-  const visibleExtent = olExtent.getIntersection(viewportExtent, layerExtent);
-
-  if (map.proj === 'geographic') {
-    if (layerExtent[1] < -180) {
-    // Previous day
-      visibleExtent[1] += 360;
-      visibleExtent[3] += 360;
-    } else if (layerExtent[1] > 180) {
-    // Next day
-      visibleExtent[1] -= 360;
-      visibleExtent[3] -= 360;
-    }
-  }
-
-  // eslint-disable-next-line no-restricted-globals
-  if (!isFinite(visibleExtent[0])) {
-    return null;
-  }
-  return visibleExtent;
-}
+let preloadMap;
 
 /**
  * Returns a promise of the layer tilegrid.
  *
  * @method promiseTileLayer
  * @param  {object} layer      _ol_layer_Tile_
- * @param  {array} extent      An array of map boundaries [180, -90, 250, 90]
  * @param  {object} viewState  Contains center, projection, resolution, rotation and zoom parameters
  * @param  {number} pixelRatio The window.devicePixelRatio, used to detect retina displays
  * @return {object}            promise
  */
-function promiseTileLayer(layer, extent, map) {
-  let tileSource;
-  let currentZ;
-  let i;
-  let tileGrid;
+function promiseTileLayer(layer, map) {
+  let i = 0;
+  return new Promise((resolve) => {
+    if (!preloadMap) {
+      const mapContainerEl = document.getElementById('wv-map');
+      const mapEl = document.createElement('div');
+      const id = 'wv-map-preload';
 
-  return new Promise((resolve, reject) => {
-    if (!extent) {
-      resolve('resolve tile layer');
-      return;
-    }
-    // OL object describing the current map frame
-    const { pixelRatio, viewState } = map.frameState_;
-    const { projection, resolution } = viewState;
-    const { zDirection } = new OlRendererCanvasTileLayer(layer);
-    tileSource = layer.getSource();
-    tileGrid = tileSource.getTileGridForProjection(projection);
-    currentZ = tileGrid.getZForResolution(resolution, zDirection);
-    i = 0;
+      mapEl.setAttribute('id', id);
+      mapEl.style.display = 'none';
+      mapContainerEl.insertAdjacentElement('afterbegin', mapEl);
 
+      preloadMap = new OlMap({
+        view: map.getView(),
+        target: id,
+      });
+    }
 
     const onLoad = function onLoad (e) {
-      if (e.type === 'tileloadend') {
-        i -= 1;
-        if (i === 0) {
-          // eslint-disable-next-line no-use-before-define
-          complete();
-        }
-      } else {
-        // eslint-disable-next-line no-console
-        console.error(`No response for tile request ${layer.wv.key}`);
-        // some gibs data is not accurate and rejecting here
-        // will break the animation if tile doesn't exist
-        // eslint-disable-next-line no-use-before-define
-        complete();
+      i -= 1;
+      preloadMap.removeLayer(layer);
```

Comment:
I think I understand this but can you add some documentation around why we're adding a layer to the preloadMap just to remove it again once loaded?